firebase-database/src/testUtil/java/com/google/firebase/database/DeepEquals.java

```diff
@@ -99,7 +100,7 @@ private static boolean weakTypeMatch(Object a, Object b) {
   }
 
   private static boolean deepEquals(Object a, Object b, Set visited) {
-    LinkedList<DualKey> stack = new LinkedList<DualKey>();
+    Deque<DualKey> stack = new ArrayDeque<>();
```

Comment:
`LinkedList` is generally prohibited because it almost never outperforms an `ArrayList` (there's a great C++ talk from Meyers on this; std::linked_list performs worse than std::vector for random operations and the gap actually increases as the list size increases. Memory locality is more important than O-notation)